webapp/Connector/src/app/store/Study.js

```diff
@@ -341,27 +356,64 @@ Ext.define('Connector.app.store.Study', {
                             return doc.hasPermission === true
                 }).length > 0;
 
-                //non-integrated assay that's in cds.assays, document, and studydocument
+                //non-integrated assay with potentially downloadable data, which may or may not also have a learn assay page
                 study.non_integrated_assay_data = documents.filter(function (doc) {
                     return doc.label && doc.docType === 'Non-Integrated Assay';
                 });
 
-                //non-integrated assay that's in studyassay
+                //non-integrated assay that has subject metadata in cds.studyassay, which may or may not also have a learn assay page
                 Ext.each(nonIntegratedAssays, function(niAssay) {
-                    var nonIntegratedAssay = {
-                        id: undefined,
-                        label: niAssay.data_id,
-                        fileName: undefined,
-                        docType: undefined,
-                        isLinkValid: undefined,
-                        suffix: undefined,
-                        sortIndex: undefined,
-                        filePath: undefined,
-                        hasPermission: niAssay.has_access,
-                        assayIdentifier: niAssay.data_id,
-                        hasAssayData: niAssay.has_data
-                    };
-                    study.non_integrated_assay_data.push(nonIntegratedAssay);
+
```

Comment:
a more efficient way would be to iterate nonIntegratedAssays once to build a map<assayIdentifier, nonIntegratedAssay>

Reply:
good suggestion, much cleaner!
done!